compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -323,9 +323,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     )
                 }
                 ExprKind::Yield(opt_expr) => self.lower_expr_yield(e.span, opt_expr.as_deref()),
-                ExprKind::Err => {
-                    hir::ExprKind::Err(self.dcx().span_delayed_bug(e.span, "lowered ExprKind::Err"))
-                }
+                ExprKind::Err => hir::ExprKind::Err(self.dcx().has_errors().unwrap()),
```

Comment:
This seems like what should actually happen is adding a `ErrorGuaranteed` tag to `ast::ExprKind::Err`? Might be worth a FIXME somewhere.

Reply:
I've done it for `ast::TyKind::Err` in #121072. `ast::ExprKind::Err` is harder, due to more unusual cases involving recovery in the parser, so I haven't finished that one yet.

Reply:
#120586 impls that, I just need to finish reviewing it 😣